layouts/partials/announcement.html

```diff
@@ -1,14 +1,29 @@
-{{ if .Page.Param "announcement" }}
-<section lang="en" id="announcement" style="background-color:{{ .Page.Param "announcement_bg" }}">
+{{ $announcementShown := false }}
+
+{{ $announcements := $.Site.Data.announcements }}
+{{ range sort $announcements "spec.priority" "desc" }}
```

Comment:
@sftim, Some initial feedback/quesitons about the data and how to use this content:

* You envision a number of unique, reusable announcement files stored as data. Each file must be parsed in these
partial templates in order to find the single announcement to publish? Any contributor can create a data file and have their announcement published based on the dates set in the file?
* Do you need to set a priority? Seems complicated and possibly error prone. Could you use a boolean (publish = true)? How many announcements can you publish at one time?
* It might be nice to have a base data file for all announcements and then have other announcements
override fields as needed.
* I see an overlap of code in the two partial templates. Could you write another template that is used by both templates?
* If these are defaults, `style="background-color: #3371e3; color: #fff;`, could add to a base (announcement) data file?
* How will the announcement be removed? Could the announcement have start and expiration dates (notAfter, notBefore?) When an
announcement expires, the "default" or base announcement (could be nothing) is displayed.
* Can the announcement message be localized?
* Will a periodic job read data files (or some other job configuration files) to determine what/when to publish (blog page)?

Reply:
> You envision a number of unique, reusable announcement files stored as data. Each file must be parsed in these partial templates in order to find the single announcement to publish? Any contributor can create a data file and have their announcement published based on the dates set in the file?

I could add a note to the website contributor guide about how to set and review announcements. Maybe that'd work.
*Only the steering committee* can approve these announcements (see issue #25759 for details). This PR doesn't implement any enforcement mechanism for details.

> Do you need to set a priority? Seems complicated and possibly error prone.

We have had announcements override other announcements (eg: KubeCon, overriding the current banner on the topic of Black Lives Matter). To me it seems like the override case is easier than the alternative: splitting the outer announcement into two and having three defined announcements that just avoid overlapping. 

> It might be nice to have a base data file for all announcements and then have other announcements
override fields as needed.

There's one required field, `.spec.message`. I don't think there's a sensible default to set for that.
As with the current implementation, the default is not to show an announcement.

> I see an overlap of code in the two partial templates. Could you write another template that is used by both templates?

I explained my thinking in https://github.com/kubernetes/website/pull/25769#discussion_r547544855

> If these are defaults, style="background-color: #3371e3; color: #fff;, could add to a base (announcement) data file?

That sounds like more complexity for a really niche case. I envisage an implementation of that covering an override mechanism for every other field to ensure a override mechanism for one. I don't think it's justified, especially as CSS is the easiest field to define overrides for (it's kind of built-in).

> How will the announcement be removed?

Removal is optional; it'll improve site build time but I suspect not by much.

> Could the announcement have start and expiration dates? Is that described in the fields: `notAfter` and `notBefore`?

Yes. Those fields are optional.

> When an announcement expires, the "default" or base announcement (could be nothing) is displayed.

As with the current implementation, the default is not to show an announcement.

> Can the announcement message be localized?

I haven't implemented this. Instead, as with the current implementation, we mark the HTML elements for the announcement as English (overriding any page-level setting, and helping any web crawlers that care).

> Will a periodic job read data files (or some other job configuration files) to determine what/when to publish (blog page)?

For blog articles, Hugo has built-in support for not publishing articles before their publication date. PR #25771 implements an automatic build trigger.
Those two mechanisms combined would make blog articles go live when ready (we use a similar mechanism for https://www.scalefactory.com/blog/).
This PR together with #25771 would make announcements go live or disappear as needed. See that PR for details about the approach.

Reply:
reverse-sorting by start date seems like it would be a simpler than a separate priority field, and I think it would do what you want in the case of a long-lived announcement being temporarily overridden by a short-lived announcement

> Can the announcement message be localized?

I also had this question, though the current localization had very limited utilization (only korean localized it, looks like), and the short-lived nature of announcements seem unlikely to mesh well with slower translation cycles.

Reply:
> reverse-sorting by start date

There could be some gotchas around sorting on start date; Hugo date parsing and handling is there, but it is a bit raw. New Hugo releases are adding useful time and date features so the topic is definitely getting attention.

To be honest I already implemented priority and I known that'll work. The start date thing seems like it could be a surprise to people who want to work out which announcement will show when a current one elapses, whereas a priority value is easy to teach.

Reply:
ok, I don't feel strongly. I don't really expect there to be more than 1-2 at any given time

Reply:
If you remove the unused fields (api, kind, spec), then `with .spec` can be removed?
I'd like to see the priority field used in practice. I'm inclined to instead use a combination of a publish flag and the time constraints.

Reply:
I guess the site will not substitute an "empty" section when there are no announcements?

Reply:
@kbhawkey examples of the `priority` field:
```yaml
---
priority: 10
dateConstraints:
  startTime: 2020-01-01T00:00:00
  endTime: 2022-07-01T00:00:00
message: |-
  Kubernetes all through 2020 _and_ the first half of 2021.  
```
and
```yaml
---
priority: 20
dateConstraints:
  startTime: 2021-03-20T09:00:00
  endTime: 2021-04-02T00:00:00
style: >-
  color: #fff; background: #505;
title: KubeCon Offworld 2021
message: |-
  Kubecon Offworld 2021 will be held on the planet Marklar from the first of April
  to the first of April.
```

Reply:
The Kubecon banner would override the “Kubernetes all through 2020 _and_ the first half of 2021.” banner for the configured duration.

Reply:
> I guess the site will not substitute an "empty" section when there are no announcements?

If there is no active announcement then there's no `section#announcement` emitted; that's the same as what the current mechanism does.

Reply:
Yes, I understand that you want to sort by priority instead of looking for a "publish" field.